src/rust/engine/process_executor/src/remote.rs

```diff
@@ -1,23 +1,57 @@
 use std::error::Error;
+use std::sync::Arc;
 
 use bazel_protos;
 use crypto::digest::Digest;
 use crypto::sha2::Sha256;
+use grpcio::{ChannelBuilder, Environment};
 use protobuf;
+use protobuf::Message;
 
 use super::{ExecuteProcessRequest, ExecuteProcessResult};
 
 ///
 /// Runs a command via a gRPC service implementing the Bazel Remote Execution API
 /// (https://docs.google.com/document/d/1AaGk7fOPByEvpAbqeXIyE8HX_A3_axxNnvroblTZ_6s/edit).
 ///
-pub fn run_command_remote(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+pub fn run_command_remote(addr: &str, req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+  let execute_request = make_execute_request(&req)?;
+
+  let env = Arc::new(Environment::new(1));
+  let channel = || {
+    ChannelBuilder::new(Arc::clone(&env)).connect(addr)
+  };
+  let execution_client = bazel_protos::remote_execution_grpc::ExecutionClient::new(channel());
+
+  let result = execution_client.execute(execute_request).unwrap();
```

Comment:
Rather than `unwrap` here, maybe:
```
let result = execution_client.execute(execute_request)?;
```

If the type of the error doesn't match, you can map_err before the ?:
https://github.com/pantsbuild/pants/blob/172d737c4848069f111a77123528ec163a8338ab/src/rust/engine/src/fs.rs#L166-L168

There are a few other `unwrap` instances in here that would be good to kill as well.

Reply:
Yeah, I was going to do that in the next PR, but might as well do it now.